Digital Twin tool crashing the engine

Hello everyone,
I saw with great enthusiasm that CARLA 0.9.15 comes with procedural environment generation based on osm data. 
I tried to follow the tutorial but the Engine crashes during environment generation.
The XODR file is generated, as well as the tiles (that become available after the crash once engine is restarted), but nothing is present in the tiles when I run manual_control.py
Am I missing something here ?

Note: I run the Widget from editor mode, which I guess is the right way to go.

---------------

CARLA version: 0.9.15
Platform/OS: Ubuntu 22.04, 24GB Ram, 
Problem you have experienced: Crash following [digital twin tutorial](https://carla.readthedocs.io/en/latest/adv_digital_twin/)
What you expected to happen: Get a procedurally generated map based on selected OpenStreetmap area
Steps to reproduce: Follow the tutorial trying to import a [test road](https://www.openstreetmap.org/export#map=15/43.7285/3.3286) ([overpassAPI link](https://overpass-api.de/api/map?bbox=3.3192,43.7197,3.3384,43.7354))

----------------
Step 1

<img src="https://github.com/carla-simulator/carla/assets/107474213/c7c6c9d6-d73e-46d1-b83d-4874c8b16d05" alt="image" width="600" height="400">

----------------
Step 2

Tried this with several OpenStreetmap locations in different countries. Here: ([overpassAPI link](https://overpass-api.de/api/map?bbox=3.3192,43.7197,3.3384,43.7354))

<img src="https://github.com/carla-simulator/carla/assets/107474213/26701c5d-453a-4b1d-8fd4-5decdfa62eab" alt="image" width="600" height="400">
Click Generate

----------------
Crash

<img src="https://github.com/carla-simulator/carla/assets/107474213/f741f4ce-f4eb-4e1a-85ef-f526eca572a9" alt="image" width="600" height="400">

```
FileName TestMap, AssetName TestMap .
[None] PostInit
=== Critical error: ===
Unhandled Exception: SIGSEGV: invalid attempt to read memory at address 0x0000000000000000

Fatal error!

0x00007f2f2403b748 libUE4Editor-Engine.so!UMaterialInterface::AnalyzeMaterialProperty(EMaterialProperty, int&, bool&) [/data/UE_4.26_carla/Engine/Source/Runtime/Engine/Private/Materials/MaterialShared.cpp:3435]
0x00007f2e400e93fb libUE4Editor-MeshMergeUtilities.so!FMeshMergeUtilities::DetermineMaterialVertexDataUsage(TArray<bool, TSizedDefaultAllocator<32> >&, TArray<UMaterialInterface*, TSizedDefaultAllocator<32> > const&, UMaterialOptions const*) const [/data/UE_4.26_carla/Engine/Source/Developer/MeshMergeUtilities/Private/MeshMergeUtilities.cpp:519]
0x00007f2e400f5b04 libUE4Editor-MeshMergeUtilities.so!FMeshMergeUtilities::MergeComponentsToStaticMesh(TArray<UPrimitiveComponent*, TSizedDefaultAllocator<32> > const&, UWorld*, FMeshMergingSettings const&, UMaterialInterface*, UPackage*, FString const&, TArray<UObject*, TSizedDefaultAllocator<32> >&, FVector&, float, bool) const [/data/UE_4.26_carla/Engine/Source/Developer/MeshMergeUtilities/Private/MeshMergeUtilities.cpp:2079]
0x00007f2dceba5d81 libUE4Editor-EditorScriptingUtilities.so!UEditorLevelLibrary::MergeStaticMeshActors(TArray<AStaticMeshActor*, TSizedDefaultAllocator<32> > const&, FEditorScriptingMergeStaticMeshActorsOptions const&, AStaticMeshActor*&) [/data/UE_4.26_carla/Engine/Plugins/Editor/EditorScriptingUtilities/Source/EditorScriptingUtilities/Private/EditorLevelLibrary.cpp:1338]
```

------------- 

Here is the [full crash report](https://github.com/carla-simulator/carla/files/13364547/crashinfo-CarlaUE4-pid-53445-37F3D5068F85497DA3557AC2CBCB7713.zip)

Other information (documentation you consulted, workarounds you tried):
* Unreal and Carla fresh build
* Tried both `Using a URL` and `Downloading an OSM file and navigating in the interface` from tutorial
* Seems there is some problems with Walker animations not found as well, I don't know if this is related somehow. I updated my content folder and am now at 20231108_c5101a5 which corresponds to version defined in `Util/ContentVersions.txt`
*  Tried filtering the osm file to only keep the roads using `osmfilter`
* Tried to generate really small zones

Thanks for your help !

I have tried to go through the whole digital twin tools in Carla 0.9.15 version and I also faced many compile and link issues when built the osmrenderer and CarlaToolsContent/OnroadMapGenerator. However, I figure them out. Here is a list of corresponding solutions: 
1. make osmrenderer: This one needs dependencies and built libosmscout and lunasvg packages. The configuration file locates in ${CARLA_BUILD_FOLDER}/Util/BuildTools/BuildOSMRenderer.sh. In this .sh file, you can see the libosmscout with specific version on Github can be cloned. 

2. But for lunasvg, it will clone the lastest version from Github. 

- This one will cause the error "`error: ‘class lunasvg::Bitmap’ has no member named ‘valid`’". The reason is the lastest lunasvg has been removed this function from its source code, replacing with isNull(). To figure this issue, it is better to manually colne the [v.2.3.8](https://github.com/sammycage/lunasvg/releases/tag/v2.3.8) to the ${CARLA_BUILD_FOLDER}/lunasvg-source (create the folder by yourself). And the comment these lines in .sh file: 
> #echo "Cloning luna-svg"
> #if [ ! -d ${LUNASVG_SOURCE_FOLDER} ] ; then
> #git clone ${LUNASVG_REPO} ${LUNASVG_SOURCE_FOLDER}
> #fi

After this lunasvg can be built smoothly. 
For Authors: would you please add the specific lunasvg COMMIT code in this .sh file? This can avoid the has no member issue. 

3. When built osm-map-renderer, it is suggested to run make PythonAPI firstly. Because the boost-1.80.0-c10-install is needed for building osm-map-renderer. Also, please modify the 

```
cmake -DCMAKE_CXX_FLAGS="-std=c++17 -g -pthread -I${CARLA_BUILD_FOLDER}/boost-1.80.0-c10-install/include" \
    ${OSM_RENDERER_SOURCE}
```
to 
 ```
cmake -DCMAKE_CXX_FLAGS="-std=c++17 -g -pthread -I${CARLA_BUILD_FOLDER}/boost-1.80.0-c10-install/include" \
-DCMAKE_EXE_LINKER_FLAGS="-L${CARLA_BUILD_FOLDER}/boost-1.80.0-c10-install/lib -lboost_system"\
    ${OSM_RENDERER_SOURCE}
```
to link the .a .so file correspondingly. 
4. The issue "usr/include/boost/system/error_code.hpp:206: undefined reference to boost::system::generic_category()'" could be raised during osmrenderer buit. Please add this command in ${OSM_RENDERER_SOURCE}/CMakeLists.txt: add_definitions(-DBOOST_ERROR_CODE_HEADER_ONLY).
5. When osmrenderer is correctly built, do make PythonAPI. 
6. After launch CarlaUE4, you can use CarlaToolsContent/OnroadMapGenerator to generate the whole scene, including buildings. PLEASE MAKE SURE THERE NO SPECIFIC SYMBOL LIKE &|* IN YOUR DOWNLOADED OSM FILE. If these symbols exist please remove or modify them in the OSM file. 
7. After this, run ./build.sh <osm_dir> in {CARLA_ROOT}/Build/libosmcout-source/maps/ to cook the osm file. Then, open UW_OnRoadMainWidget in Unreal and copy the file dir into Select OSM Database, and Please DO NOT CLICK any Button, otherwise the Unreal will crash. 
8. Open a new cmd under {CARLA_ROOT}/Build/osm-world-renderer-build and run ./osm-world-renderer. Until you see "Starting renderer C
┌ Waiting Command.."
9. Click Config and Render in UW_OnRoadMainWidget, also you need to input the geoloaction in Navigation Settings, which can be checked in OSM webpage. Then, you can see the osm map rendered.
10. IMPORTANT: after all setting, please input you File Name, and input you downloaded and modified local .osm file in XODR OR OSM FilePath. Then, click the Generate and have a cup of coffee. 
11. Enjoy the scene you created. Here is my example: 
![Screenshot from 2024-09-27 15-38-12](https://github.com/user-attachments/assets/a3bff296-5c85-466a-bb19-b315520b94f3)


```

--- a/Util/BuildTools/BuildOSMRenderer.sh
+++ b/Util/BuildTools/BuildOSMRenderer.sh
@@ -39,7 +39,7 @@ cd ${LIBOSMSCOUT_BUILD_FOLDER}
 cmake ${LIBOSMSCOUT_SOURCE_FOLDER} \
     -DCMAKE_INSTALL_PREFIX=${INSTALLATION_PATH}
 
-make
+make -j$(nproc)
 make install
 
 # ==============================================================================
@@ -47,7 +47,7 @@ make install
 # ==============================================================================
 echo "Cloning luna-svg"
 if [ ! -d ${LUNASVG_SOURCE_FOLDER} ] ; then
-  git clone ${LUNASVG_REPO} ${LUNASVG_SOURCE_FOLDER}
+  git clone -b v2.3.8 --depth 1 ${LUNASVG_REPO} ${LUNASVG_SOURCE_FOLDER}
 fi
 
 mkdir -p ${LUNASVG_BUILD_FOLDER}
@@ -56,7 +56,7 @@ cd ${LUNASVG_BUILD_FOLDER}
 cmake ${LUNASVG_SOURCE_FOLDER} \
     -DCMAKE_INSTALL_PREFIX=${INSTALLATION_PATH}
 
-make
+make -j$(nproc)
 make install
 
 
@@ -69,7 +69,8 @@ mkdir -p ${OSM_RENDERER_BUILD}
 cd ${OSM_RENDERER_BUILD}
 
 cmake -DCMAKE_CXX_FLAGS="-std=c++17 -g -pthread -I${CARLA_BUILD_FOLDER}/boost-1.84.0-c10-install/include" \
+-DCMAKE_EXE_LINKER_FLAGS="-L${CARLA_BUILD_FOLDER}/boost-1.84.0-c10-install/lib -lboost_system" \
     ${OSM_RENDERER_SOURCE}
-make
+make -j$(nproc)
 
 echo "SUCCESS! Finishing setting up renderer."

